### PR TITLE
Added php-opencloud driver

### PR DESCRIPTION
**Pull request by kynx**
As discussed in #266, this adds a driver using php-opencloud.

### CODE REVIEW

**Comment by kynx** [comment]
@henrikbjorn Don't want to nag, but any thoughts on this?

File: src/OpenCloudDriver.php [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+
+    /**
+     * {@inheritDoc}
+     */
+    public function listQueues()
+    {
+        $result = [];
+        /* @var $queues Queue[] */
+        if ($queues = $this->service->listQueues()) {
+            foreach ($queues as $queue) {
```
Comment (henrikbjorn) [inline src/OpenCloudDriver.php:71]: maybe use array_map here.

File: src/OpenCloudDriver.php [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+                    $this->cache->push($queueName, [$claim->getBody(), $claim->getHref()]);
+                }
+
+                if (count($claims)) {
+                    return $this->cache->pop($queueName);
+                }
+            }
+
+            //sleep for 10 ms
+            usleep(1000);
```
Comment (henrikbjorn) [inline src/OpenCloudDriver.php]: Should use interval.

File: src/OpenCloudDriver.php [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+    /**
+     * {@inheritDoc}
+     */
+    public function removeQueue($queueName)
+    {
+        try {
+            $queue = $this->getQueue($queueName, false);
+            $queue->delete();
+        } catch (BadResponseException $e) {
+            if ($e->getResponse()->getStatusCode() != 404) {
```
Comment (henrikbjorn) [inline src/OpenCloudDriver.php]: Isnt it an error to remove a non existant queue?

File: src/OpenCloudDriver.php [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+
+    /**
+     * Returns queue, optionally creating it if it doesn't already exist
+     * @param string $name
+     * @param bool $create
+     * @return Queue
+     */
+    private function getQueue($name, $create = true)
+    {
+        if (empty($this->queues[$name])) {
```
Comment (henrikbjorn) [inline src/OpenCloudDriver.php]: if would change the order of the if, so we can skip an indentation.

```
if (!$this->queues[$name]) {
   return $this->queues[$name];
}
```

**Review by henrikbjorn — APPROVED** [review]
I like the guzzle mock stuff. Very awesome.

File: src/OpenCloudDriver.php (hunk 1 above)
Comment (kynx) [reply src/OpenCloudDriver.php:71]: Unfortunately `listQueues()` returns an Iterator, so array_map won't work.

File: src/OpenCloudDriver.php (hunk 2 above)
Comment (kynx) [reply src/OpenCloudDriver.php]: OK, but wouldn't that mean it'll only ever loop once? The Doctrine, Mongo and Amqp drivers all `usleep(10000)`.

File: src/OpenCloudDriver.php (hunk 3 above)
Comment (kynx) [reply src/OpenCloudDriver.php]: OK. On a side-note, it'd be good to have some standard exceptions for users to catch in cases like this.

**Comment by kynx** [comment]
@henrikbjorn  Oops, the commit ate my comments ;)

Regarding `usleep()` in popMessage() I'm kinda confused. The Doctrine, Mongo and Amqp drivers all usleep(10000). Wouldn't sleeping for `$interval` mean it'll only ever loop once?

**Pull request merged**